include/nanorange/detail/memory/temporary_vector.hpp

```diff
@@ -0,0 +1,94 @@
+
+// nanorange/detail/memory/temporary_vector.hpp
+//
+// Copyright (c) 2019 Tristan Brindle (tcbrindle at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef NANORANGE_DETAIL_MEMORY_TEMPORARY_VECTOR_HPP_INCLUDED
+#define NANORANGE_DETAIL_MEMORY_TEMPORARY_VECTOR_HPP_INCLUDED
+
+#include <nanorange/memory/destroy.hpp>
+
+#include <cassert>
+#include <memory>
+#include <new>
+
+NANO_BEGIN_NAMESPACE
+
+namespace detail {
+
+template <typename T>
+struct temporary_vector {
+private:
+    struct deleter {
+        void operator()(T* ptr) const { ::operator delete[](ptr); }
+    };
+
+public:
+    temporary_vector() = default;
+
+    explicit temporary_vector(std::size_t capacity)
+        : start_(static_cast<T*>(::operator new[](capacity * sizeof(T), std::nothrow))),
+          end_cap_(start_ ? start_.get() + capacity : nullptr)
+    {}
+
+    temporary_vector(temporary_vector&& other) noexcept
+        : start_(std::move(other.start_)),
+          end_(other.end_),
+          end_cap_(other.end_cap_)
+    {
+        other.start_.reset();
```

Comment:
This `reset` is redundant: a moved-from `unique_ptr` is guaranteed to hold `nullptr`.

Reply:
I was 99% sure that was the case, but I added the `reset()` out of paranoia